gap/SkeletalFinSetsForCAP.gi

```diff
@@ -4,6 +4,23 @@
 # Implementations
 #
 
+##
+AddDerivationToCAP( MorphismsOfExternalHom,
+  function ( cat, A, B )
+    local hom_A_B, D;
+    
+    hom_A_B := HomomorphismStructureOnObjects( cat, A, B );
+    
+    D := DistinguishedObjectOfHomomorphismStructure( cat );
+    
+    return List( hom_A_B,
+                 i -> InterpretMorphismFromDistinguishedObjectToHomomorphismStructureAsMorphism( cat,
+                         A,
+                         B,
+                         MapOfFinSets( D, [ i ], hom_A_B ) ) );
```

Comment:
Is there a reason why `D = DistinguishedObjectOfHomomorphismStructure( cat )` always has to be the terminal object?

I try to produce a counter-example: Let `cat` be the subcategory of `SkeletalFinSets` with the single object 2 and only the identity in 2. Choose $D = 2$ and $H(2,2) = 1$.